### DISCUSSION TITLE
Possible to get JSON schema type map from LiteLLM?

### DISCUSSION THREAD

**Discussion by jamesbraza** [BerriAI/litellm#3531] [Q&A]
I have defined the below type map to convert Python types to JSON schema types. This is used to create tool signatures.

```python
# Mapping from python types to JSON schema types
# SEE: https://json-schema.org/understanding-json-schema/reference/numeric
type_map: dict[type, str] = {
    str: "string",
    int: "integer",
    float: "number",
    bool: "boolean",
    list: "list",
    dict: "object",
}
```

Is there some way I can import something like this from LiteLLM?

**Comment by ishaan-jaff** [comment]
@jamesbraza I did not understand what you need Can you elaborate  / show a code snippet of what you want from litellm ?

**Reply by jamesbraza** [reply]
My use case is a Python function that I have both type hinted with a Google style docstring. I made code to inspect the function (using `inspect`) and convert it into a JSON tool schema that I can pass to LiteLLM.

The type hints, I use the above `type_map` to convert the type hints (e.g. `float`) into JSON schema types (e.g. `number`).

I am wondering, is there some LiteLLM built-in that can execute this conversion for me?
